sklearn/svm/_base.py

```diff
@@ -335,7 +332,7 @@ def _dense_fit(self, X, y, sample_weight, solver_type, kernel, random_seed):
             y,
             svm_type=solver_type,
             sample_weight=sample_weight,
-            class_weight=self.class_weight_,
+            class_weight=self._class_weight,
```

Comment:
Do you think this is easier to follow or using `@property`?

```python
class_weight=getattr(self, "_class_weight", np.empty(0))
```

For me, it's about the same.

Reply:
I agree with @thomasjpfan's suggestions since this simplify the logic not using overridden properties.

Taking your comment in again:
```suggestion
            # Class weights are defined for classifiers during
            # fit, otherwise it is empty.
            class_weight=getattr(self, "_class_weight", np.empty(0)),
```

What do you think?

Reply:
I'm not sure I quite understand this. The original intention for this approach was to ensure there wouldn't be a `FutureWarning` raised during `fit` or `predict` for `SVR/NuSVR/OneClassSVM` but using `getattr` instead does raise that warning. So then we'd either have to suppress the warning locally, or keep the current form and I prefer the latter.

I may very well just be misunderstanding the suggestion though, so feedback welcome :)

Reply:
Not sure I understand. The use of `_class_weight` is to ensure `fit` and `predict` don't raise a `FutureWarning`, but using the proposed approach w/ `getattr` still raises it, therefore we'd need to suppress it locally (in 5 places) which seems messy. I may just be misunderstanding the suggestion though 😅